tests/net/all/src/main.c

```diff
@@ -18,23 +18,31 @@ LOG_MODULE_REGISTER(net_test, LOG_LEVEL_DBG);
 #include <zephyr/net/net_if.h>
```

Comment:
Basically LTGM. Small nit about the tests, was the idea that the tests would be in the second commit? Now the first commit contains also some test code.

Reply:
Yes, that was the idea

The changes to `tests/net/all` and `tests/net/socket/offload_dispatcher` are just minor tweaks to ensure both tests still build and run successfully with the new changes, whereas the second test introduces an entire new test suite

The reason I added a second net device to `tests/net/all` is that originally that test was using a net device that was simultaneously `offloaded` and `dummy`. It is no longer possible to have a device be both simultaneously, so I instanced one of each